test/e2e_node/summary_test.go

```diff
@@ -53,9 +55,10 @@ var _ = framework.KubeDescribe("Summary API", func() {
 			const pod1 = "stats-busybox-1"
 
 			By("Creating test pods")
-			createSummaryTestPods(f, pod0, pod1)
-			// Wait for cAdvisor to collect 2 stats points
-			time.Sleep(15 * time.Second)
+			pods := getSummaryTestPods(f, pod0, pod1)
```

Comment:
To match the containers in a pod, we use the [MatchAllElements](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/summary_test.go#L128) method of gstruct.  This matches elements in a list, with one match allowed for each.  See the [tests for gstruct](https://github.com/onsi/gomega/blob/master/gstruct/elements_test.go#L25).  So we were already checking for duplicate elements before.  We just never tested on a restarted container.